src/string.cr

```diff
@@ -3377,24 +3404,33 @@ class String
       char_index += 1
     end
 
-    head_pointer = pointer
+    return if pointer + search.bytesize > end_pointer
 
-    # calculate a rolling hash of this text (haystack)
+    if search.bytesize <= 8
+      search_bytesize = search.bytesize
+      return index_short(UInt64, char_index, pointer, end_pointer, search) { |pointer|
```

Comment:
I don't think the compiler source has a code style defined anywhere besides `crystal tool format`, but usually `{...}` is used on inline blocks and `do/end` on multi line blocks. Since you use `do/end` on all other blocks in this patch it would be good to use it here too for the sake of consistency.